src/rgw/rgw_op.cc

```diff
@@ -3659,7 +3659,7 @@ int RGWDeleteObj::handle_slo_manifest(bufferlist& bl)
 
 int RGWDeleteObj::verify_permission()
 {
-  if (!verify_bucket_permission(s, RGW_PERM_WRITE)) {
+  if (!verify_bucket_permission(s, RGW_PERM_WRITE) && s->user->user_id.compare(s->bucket_owner.get_id()) != 0) {
```

Comment:
this change in `RGWDeleteObj` is not actually needed - we're only rejecting the HEAD request with '403: Forbidden'. `verify_bucket_permission()` here is correctly detecting the owner's `RGW_PERM_WRITE` permission